lib/watir/element_collection.rb

```diff
@@ -162,6 +161,19 @@ def elements
       end
     end
 
+    def elements_with_tags
+      els = elements
+      if @selector[:tag_name]
+        els.map { |e| [e, @selector[:tag_name]] }
+      else
+        begin
+          els.zip(browser.wd.execute_script('return arguments[0].map(function(e) {return e.localName});', els))
+        rescue Selenium::WebDriver::Error::ObsoleteElementError
+          elements_with_tags
```

Comment:
It's unlikely, but if it happened there's no useful information about what's going on. This is how I addressed it elsewhere: https://github.com/watir/watir/blob/master/lib/watir/locators/element/locator.rb#L200-L212
Whenever I get back around to the Watir Executor, this can be cleaned up.

Reply:
well this link is super unuseful  after the refactor

Reply:
I'll take a look again after everything stops moving. I know what you're referring to https://github.com/watir/watir/blob/1e1f7f891bfce7aa4acae6bf36490f7994923fa7/lib/watir/locators/element/locator.rb#L40-L51

Reply:
Ugh, really we want to wrap these in an `#element_call` type method that respects the global timeout. I think this is where the Watir::Executor ideally gets really handy.